src/hqdn3d.cpp

```diff
@@ -20,48 +20,48 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <cmath>
+#include <cstdlib>
 #include <cstdint>
 #include <algorithm>
 
-#include <vapoursynth/VapourSynth.h>
-#include <vapoursynth/VSHelper.h>
+#include <VapourSynth.h>
+#include <VSHelper.h>
 
 typedef struct Hqdn3dData {
     VSNodeRef *clip;
     const VSVideoInfo *vi;
 
-    double lumSpac   = -1.0;
-    double chromSpac = -1.0;
-    double lumTmp    = -1.0;
-    double chromTmp  = -1.0;
-    int restartLap   = -1;
+    double lumSpac;
+    double chromSpac;
+    double lumTmp;
+    double chromTmp;
+    int restartLap;
 
     int coefs[4][512*16];
-    int planeCnt     = -1;
-    unsigned int ** prevFrame = nullptr;
-    unsigned int ** prevLine  = nullptr;
+    unsigned int *prevFrame[3] = { 0 };
```

Comment:
Not all color definitions make use of 3 planes. Therefor I've preferred, to keep it as a general solution

Reply:
When the input clip has only one plane the two remaining elements of the arrays are wasted, but that's just 32 bytes on 64 bit systems. On the other hand, you get rid of two malloc/free pairs.